### ISSUE TITLE
ABC iview and SBS on Demand

### ISSUE THREAD

**Issue by dawsonjw** [matthuisman/slyguy.addons#737]
Hi
     Just wondering if you had any plans to add ABC iView and SBS on demand to the stable as the current AussieAddons ones are no longer supported. 
     
Oh thanks for the great addons you currently maintain. 

cheers

John

**Comment by matthuisman** [comment]
i would like too eventually if AussieAddons has stopped supporting it which I think may be the case.
They also may just be busy with life which happens to all of us at times

@glennguy - how would you feel about this?

**Comment by dawsonjw** [comment]
Hi.  I am pretty sure if you go to the web page it says at the bottom of the page it has been shutdown. 

https://aussieaddons.com/

Hopefully you will get an answer from the author 

cheers

John

**Comment by matthuisman** [comment]
the repo was shutdown, but the addons were still being updated for some time after that

**Comment by glennguy** [comment]
Yeah the shutdown was back in 2018 after getting a cease and desist from a couple of providers. The content was removed then we came back online.

Those are the only two add-ons that are somewhat maintained but they seem to be working well without much interference. I was actually only looking two nights ago about adding a 1080p 'hack' into iview.

> @glennguy - how would you feel about this?

I spoke to Andy not too long ago and he wasn't fussed at all. He doesn't have much time these days and tbh I don't either. I'm happy to fix small issues but if there was a massive API change I don't know if I could get the time quick enough to support it.

I still want to contribute to inputstream.adaptive as well but the last 100 PRs have not been from me, I'm just there as a reviewer basically. So anyway if you wanted to create slyguy iview/sbs then you'd have our blessings of course.

**Comment by matthuisman** [comment]
cool :) i dont really have time at the moment either, but its nice to know i want step on toes if i do find time and put something out

**Comment by dawsonjw** [comment]
Thanks to all

John

**Comment by glennguy** [comment]
No worries John. Please raise an issue over on aussie addons github if there's any problems, or you can usually get some good help on the WP forums.

**Comment by sliceanddice9** [comment]
Any updates? The 'Aussie Add-On' versions of SBS On Demand and ABC iview on Kodi aren't working on my Kodi build on my phone. Would love to see Slyguy versions of SBS On Demand and ABC iview please.

**Comment by matthuisman** [comment]
https://www.abc.net.au/about/media-centre/press-releases/abc-is-no-1-for-audience-reach-across-streaming-and-broadcast-in/106101904

ABC will be the first I do as its most popular :)

**Comment by dawsonjw** [comment]
Hi Matt.

              Thanks for this.

Cheers

John

Sent from my iPhone

On 14 Jan 2026, at 2:47 pm, Matt Huisman ***@***.***> wrote:

﻿
[https://avatars.githubusercontent.com/u/6225961?s=20&v=4]matthuisman left a comment (matthuisman/slyguy.addons#737)<https://github.com/matthuisman/slyguy.addons/issues/737#issuecomment-3747560993>

https://www.abc.net.au/about/media-centre/press-releases/abc-is-no-1-for-audience-reach-across-streaming-and-broadcast-in/106101904

ABC will be the first I do as its most popular :)

**Comment by matthuisman** [comment]
Looks like SBS has made some changes breaking the existing aussie addons SBS addon
https://github.com/aussieaddons/plugin.video.sbs/issues/1304

So if that is no longer updated, I would do SBS first (instead of ABC) so users can have a working option.

**Comment by grapenose** [comment]
That would be really good.

**Comment by dawsonjw** [comment]
Hi
    Thanks for that.  Happy to test

Cheers

John

**Comment by Spin35** [comment]
I would be most grateful if you were able to direct some SlyGuy energy towards fixing the SBS On Demand add-on. It has been running sweet up until a few days ago, and I'm hoping that whatever SBS have done to the streams doesn't require major effort. I note that the SBS IPTV channels have been fixed, so hopefully this bodes well for the add-on.
TBH, I find the iView add-on a bit cumbersome to use and it has no logon, so users can't access their iView account watchlist (feature request?). Given there are no ads on the ABC, I find the iView android app does the job for me.

**Comment by Garrylua** [comment]
Hi. I would also very much appreciate a fix to the SBS On Demand addon.

**Comment by uk-julian** [comment]
Indeed. I just came here after hitting a link in the aussiesddons. Very helpful and explains why I haven't been able to access sbs.Hopefully you have the time to get addon fixed as use it a lot, but at least now know reason why it all stopped suddenly a few days ago.

**Comment by matthuisman** [comment]
Can someone with an android tv install latest sbs app and then export it to apk (use xplore or something similar)

**Comment by Tyreless** [comment]
I downloaded it - it’s a 92Mb zip file with 4 apk’s. I’m a little naive, so can you tell me the best way to get it to you?

**Comment by matthuisman** [comment]
Google drive and then share?
blog@matthuisman.nz

**Comment by Tyreless** [comment]
Shared from my synology - let me know if you are unable to access it.

**Comment by matthuisman** [comment]
Thank you @Tyreless 
Addon is in progress :)

**Comment by matthuisman** [comment]
A very basic initial SBS addon is now in the slyguy repo.
It just does the Login + Live channels +IPTV Merge compatibility

Ill work on the VOD stuff next

**Comment by dawsonjw** [comment]
Hi
    First release works like a charm.  

Thanks heaps

John

**Comment by Spin35** [comment]
Installed V0.1. It seems to be working fine. I added SBS Chill to the Bookmarks, but Bookmarks isn't visible in the entry level menu. They are enabled in the settings.
Slight UI irregularity - when I hit stop on a live-tv channel, it takes me to an empty screen with .., which in turn takes me back to the entry level main menu. I would have thought hitting stop would take you back to the channel list.
Thank you for a great effort considering how quickly you've got this out.
Looking forward to seeing the VOD in action.

**Comment by matthuisman** [comment]
oops. bookmarks entry added.

I can't reproduce the live-tv channel thing...

If I go to SBS -> Live TV -> Play Channel -> Stop 
I get the Live TV page again.
Are you playing them differently?
What skin?

**Comment by dawsonjw** [comment]
Hi
     Stopping live tv works as expected for me. Bookmark works fine now. 

John

**Comment by Spin35** [comment]
Bookmarks fix noted - cheers :-)
I'm running the Plextuary skin which is a modded version of Estuary that works with Plex.
The stopping live-tv behaviour is inconsistent; sometimes it goes to .. , other times it returns to live-tv , and otherother times it goes to live-tv but the channel list starts at France24 onwards (no SBS free-to-air channels) - if I go back to main menu and then back into live-tv the full list is there.
Stopping Play from a channel in bookmarks takes me back to bookmarks.
Not sure if it's related but sometimes the VideoOSD page only has pause and stop buttons on the left, while other times it has << and >> buttons as well. This is also inconsistent and hard to replicate.
In any case, I don't see these as serious issues, they've just observations on how my UI works, and they may be related to my setup, so don't waste any serious time on this stuff - nothing is "broken".
I think that getting the VOD going is what most folks are looking for and I wouldn't want to distract you from that.

**Comment by matthuisman** [comment]
@Spin35 Id suggest trying the standard skin and see if same behaviour
Addons dont control the kodi UI
Dont use "widgets" either - use the addon via addons menu

**Comment by matthuisman** [comment]
just pushed 0.0.3 that adds majority of VOD and "watchlist" (SBS call it favorites) support and search
There will still be issues with some news / live events I assume but majority of normal vod should work.

**Comment by dawsonjw** [comment]
Hi Matt
          Thanks for that. Everything obvious seems to work ok. Just wondering if there is a way to sort ALL the menus by NAME with one click rather than doing it manually   I know your other addon don’t but it would be handy. 

Cheers 

John

**Comment by matthuisman** [comment]
By default itll be in the same order that their website/app uses
Im not sure if kodi can change the default sort order..
Might be able to do it in addon, but would need new settings for it (something like https://github.com/matthuisman/slyguy.addons/issues/700)

**Comment by dawsonjw** [comment]
Hi
     Actually that would be fabulous but I know you are really busy.  One for the wishlist I guess. Just had a look at the ABC iview addon and it seems to manage to lock the sort order to DATE. So I guess it is possible 

Cheers 

John

**Comment by matthuisman** [comment]
Episodes should list in correct order in my sbs.
By episode number.

Which menus do you prefer ordered by name?
I assumed sbs would show most relevant first so that's why I like to keep their order by default

**Comment by matthuisman** [comment]
Also, ordering won't work across pages. The ordering is done after a page is retrieved.

**Comment by Tyreless** [comment]
Big thanks! Two requests:
1. Ordering the Watchlist would be great.
2. Subtitles don’t appear to be working yet.

**Comment by matthuisman** [comment]
Is the watchlist order different to order on sbs website?
You can sort using Kodi sort options 

Which content is confirmed to have subtitles? I'll check that and figure it out

**Comment by Tyreless** [comment]
Lots of sbs content has subtitles. Example, the movie “Point Break”.

**Comment by uk-julian** [comment]
Great work on the sbs . I tried two Nordic noir series and subtitles worked fine.

The only thing missing which I find important is the leaving date for shows / films., so as to calendar your watch list . Aussie addons had them shown, don't know how much of a job this would be, as obviously most addon don't need this specific qualification.

Cheers



Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Spin35** [comment]
Big Thanks for the added VOD on V0.0.3 - I can now continue my revisit of "The Orville" and look forward to checking out "The Audacity".
For subtitles, folks might need to check their settings in the Language area. Mine were working by default with V0.0.1 and I switched them off - the setting has carried through successive updates.

**Comment by Tyreless** [comment]
Still no luck on subtitles. I can confirm not working on, for example, Alone Australia - no subtitles in the addon but a range of subtitles in the android tv app. Hard coded subtitles (eg: Backstrom) work of course.

Checked my Language settings for subtitles in the addon. The settings are:

Subtitle Allow List: All
Include Forced Subtitles: Yes
Include Non-forced Subtitles: Yes
Default Subtitle: original,interface,en

**Comment by matthuisman** [comment]
What version Kodi and inputstream.adaptive using?

**Comment by Tyreless** [comment]
Kodi 21.2.0
Inputstream.adaptive 21.5.18

**Comment by matthuisman** [comment]
Looks like the subs are embedded in the Google DAI streams but not the Akamai HLS streams
I should be able to inject them or worse case switch to the DAI streams (but prefer to avoid these)

**Comment by matthuisman** [comment]
@Tyreless  please try 0.0.5 (just pushed)

**Comment by Tyreless** [comment]
Good news: Subtitles working!
 but…. they all need a delay of 21.7 seconds. Tried a few episodes of Alone Australia and the movie Point Break, all with the same result.

**Comment by Rossi815** [comment]
Thanks for the update. When I try to access a Euro cycling event like Paris Roubaix or Liege Bastogne Liege I get a "Failed to fetch JSON data" error. The other sporting events work fine, even cycling events like Cape epic 2026 stage 7. 
Cheers

**Comment by matthuisman** [comment]
As above, sports etc not fully supported yet

**Comment by Rossi815** [comment]
OK thanks.

**Comment by matthuisman** [comment]
I'll work on that next though. Hopefully tonight

**Comment by webaware** [comment]
Looking great, thanks Matt!

One thing that would be nice to have is the expiry dates showing on the info for a show, like the old SBS app did. Should I open a new issue for that? Very helpful for working out what needs to be watched first / before it gets pulled.

**Comment by matthuisman** [comment]
Is it just shows that expire? Or episodes or movies?

**Comment by webaware** [comment]
It's episodes and movies; I don't think shows / seasons have (or had) that info. 

e.g. [this movie expires 29 June 2026](https://www.sbs.com.au/ondemand/movie/where-the-money-is/2432549443698).

e.g. [these episodes expire 30 April 2026](https://www.sbs.com.au/ondemand/tv-series/vernon-subutex).

**Comment by djgw** [comment]
Thanks Matt, much appreciated,, working well here.
And good to know the subtitle offset, thanks @Tyreless

**Comment by matthuisman** [comment]
I'll look into subs as well. Might need to swap to the dai streams that the app/website uses

**Comment by matthuisman** [comment]
> It's episodes and movies; I don't think shows / seasons have (or had) that info.
> 
> e.g. [this movie expires 29 June 2026](https://www.sbs.com.au/ondemand/movie/where-the-money-is/2432549443698).
> 
> e.g. [these episodes expire 30 April 2026](https://www.sbs.com.au/ondemand/tv-series/vernon-subutex).

Great. Just having it in the description is ok?

**Comment by webaware** [comment]
The Aussie Addons one put it at the bottom of the description, and you can see that on their iView add-on. "Expires Wed, 30 April 2026" or similar. Very handy for letting one know that there's not much time left to finish a series!

**Comment by matthuisman** [comment]
Ok sweet. Easy 😁 as long as the changed API still returns if. If the app shows it I'm sure it does

**Comment by webaware** [comment]
Fabulous. NB: I should mention, please always show it (unlike the website and their iPhone app). It's nice to know that something expires in December 2026, or October 2028, so that viewers can organise which shows to view in what order.

I hope the new API has it still!

**Comment by matthuisman** [comment]
0.0.6 just pushed which adds Expires to the plot.

If its expiry within a month, itll show as yellow and in humanize form "Expires in 3 days" etc.
If its longer than that, itll show in white and just the date "Expires 24/01/28"

**Comment by webaware** [comment]
Thanks, looks really good on shows and movies within the add-on. Yellow near-term expiry is great, it really highlights what's going to go away soon.

It doesn't show for episodes, which can differ from episode to episode (as they can be added on different days and get different expiry dates) and season to season (older seasons expiry sooner). It would be handy if you could add to episodes too please.

With movies and shows added to Favourites, likely nothing will be seen. I tried in my selected theme (Metropolis) and the default (Estuary) and it's the same. Favourites in Kodi doesn't let you see show or movie descriptions, but it does let you see episode descriptions. Another reason it would be nice to have expiry date on individual episodes please.

Aside: If you favourite a collection of movies (like the [John Wick Trilogy](https://www.sbs.com.au/ondemand/collection/john-wick-trilogy)), you can see descriptions on the movies, with expiry dates; but if you favourite an individual movie you get no description and thus no expiry date (this is likely something you can't do anything about, it's Kodi).

**Comment by matthuisman** [comment]
Favourites doesn't bring in plot. And it's static too so it wouldn't change anyway. 

I intentionally left off episodes so will enable that

**Comment by matthuisman** [comment]
0.0.8 just pushed which adds much better support for news and sports
@Rossi815  should fix your issue

**Comment by matthuisman** [comment]
Subtitle issue is caused by inputstream.adaptive (https://github.com/xbmc/inputstream.adaptive/pull/2024)
0.0.9 pushed which disables IA and just uses ffmpeg for playback which fixes the subs

**Comment by Rossi815** [comment]
> 0.0.8 just pushed which adds much better support for news and sports [@Rossi815](https://github.com/Rossi815) should fix your issue

It did. Thanks very much.

**Comment by Tyreless** [comment]
Subtitles working now - timing is correct. Many thanks!

**Comment by Spin35** [comment]
@matthuisman I've just visited the yt-dlp site and they too have been grappling with subs.  There might be something useful here:
[https://github.com/yt-dlp/yt-dlp/issues/16529](url)

**Comment by matthuisman** [comment]
Not related. the subs issue is an inputstream.adaptive issue.
Solved by swapping to DAI streams withe embed streams otherwise dispabling IA for playbacj

**Comment by matthuisman** [comment]
Can anyone confirm if the ABC Android to app currently lets them login?
The website just keeps saying my code is wrong
"Sorry, that code doesn't match the code on your TV. Please try again."

**Comment by matthuisman** [comment]
oh, the tv says to go to
https://abc.net.au/link

And that was taking me to:
https://account.abc.net.au/device

But actually, the new url is:
https://account.abc.net.au/device-verify
(The /device is for older 8 character codes)


Weird...

**Comment by Tyreless** [comment]
I’m not having any problems logging in to the Android TV app on my Shield. I guess it was a previously linked device so it logs in automatically.

**Comment by Spin35** [comment]
I'm guessing from the posts in the last couple of hours that you have now moved on to a SlyGuy iView add-on?  Thanks for doing that, but I'd like to suggest that you take time to bask in the glory of your efforts on SBS On-Demand.  You went from V0.0.1 to 0.0.12 in less than a week and I'm wondering how many pizza boxes are surrounding your work area right now?

**Comment by uk-julian** [comment]
> On

🤣🤣🤣

**Comment by Tyreless** [comment]
Subtitles are no longer available on SBS On Demand v 0.14.

**Comment by Tyreless** [comment]
On further testing, going back to v 0.13 or 0.12 didn’t fix the subtitles for me.

**Comment by matthuisman** [comment]
Run latest, make sure "PREFER DAI streams" in the addon settings is enabled

**Comment by Tyreless** [comment]
Yep, “Prefer DAI streams” was enabled (and I have left it enabled). I have now found that subtitles work if I enable “Use Inputstream HLS” for live streams and VOD streams.

**Comment by matthuisman** [comment]
Ok thank you. What version Kodi are you?
I pushed an update to common last night the sets those to default off as having more streams that IA doesn't like in various places. I'll undo that change so they default to On

**Comment by matthuisman** [comment]
Latest common pushed that reverts that default change

**Comment by matthuisman** [comment]
I've created a new issue for the abc iview request 
https://github.com/matthuisman/slyguy.addons/issues/1218
